### PR TITLE
Project Euler Question 37

### PR DESCRIPTION
**Pull request by square-1111**
**Fixes issue:** N/A

**Changes:**  
- Added Problem #37 in C++.

### CODE REVIEW

File: code/online_challenges/src/project_euler/problem_037/37.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
+#include < bits / stdc++.h >
+    using namespace std;
```
Comment (YashovardhanSiramdas) [inline code/online_challenges/src/project_euler/problem_037/37.cpp]: No Indentation is required for ```namespace```

File: code/online_challenges/src/project_euler/problem_037/37.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+ * Sieve function finds all the primes upto n-1 using Sieve of Erasthosene method.
+ * Time Complexity = O(n logn)
+ */
+
+void sieve(long long n, bool prime[]) 
+{
+   prime[0] = prime[1] = false;
+
+   for (int i = 2; i < n; i++) 
+   {
```
Comment (YashovardhanSiramdas) [inline code/online_challenges/src/project_euler/problem_037/37.cpp]: Braces are not required if loop has only 1 line of code

File: code/online_challenges/src/project_euler/problem_037/37.cpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+       prime[i] = true;
+   }
+   for (int i = 2; i * i <= n; i++) 
+   {
```
Comment (YashovardhanSiramdas) [inline code/online_challenges/src/project_euler/problem_037/37.cpp]: Same Here. Remove braces from line numbers 18,20,23,24

File: code/online_challenges/src/project_euler/problem_037/37.cpp [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+       }
+   }
+}
+
+/*
+ * check_prime checks if a number is prime.
+ * Time complexity O(sqrt(n))
+ */
+
+bool check_prime(long long num) {
```
Comment (YashovardhanSiramdas) [inline code/online_challenges/src/project_euler/problem_037/37.cpp]: Function opening brace should be in its own line

File: code/online_challenges/src/project_euler/problem_037/37.cpp [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+   if (num == 2 || num == 3)
+       return true;
+
+   if (num == 1 || num == 0)
+       return false;
+    
+   for (int i = 3; i * i <= num; i += 2) 
+   {
```
Comment (YashovardhanSiramdas) [inline code/online_challenges/src/project_euler/problem_037/37.cpp]: Braces not required here

File: code/online_challenges/src/project_euler/problem_037/37.cpp [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+bool check_right(long long num) 
+{
+   if (check_prime(num) == false)
+       return false;
+
+   bool isprime[num + 1];
+    
+   sieve(num + 1, isprime);
+    
+   for (int i = num; i > 0; i /= 10) {
```
Comment (YashovardhanSiramdas) [inline code/online_challenges/src/project_euler/problem_037/37.cpp]: Braces not required

File: code/online_challenges/src/project_euler/problem_037/37.cpp [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    
+   return count;
+}
+
+/*
+ * check_left checks if a number is left truncatable prime number
+ *    Time Complexity O(n logn)
+ */
+
+bool check_left(long long num) {
```
Comment (YashovardhanSiramdas) [inline code/online_challenges/src/project_euler/problem_037/37.cpp]: Opening Brace should be in its own line

File: code/online_challenges/src/project_euler/problem_037/37.cpp [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+   return true;
+}
+
+/*
+ * check_two_way_truncatable checks if a number is right 
+ * as well as left truncatable.
+ */
+
+bool check_two_way_truncatable(long long num) {
```
Comment (YashovardhanSiramdas) [inline code/online_challenges/src/project_euler/problem_037/37.cpp]: Same Here

File: code/online_challenges/src/project_euler/problem_037/37.cpp [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+   return check_left(num) && check_right(num);
+}
+
+int main() {
```
Comment (YashovardhanSiramdas) [inline code/online_challenges/src/project_euler/problem_037/37.cpp]: Same here

File: code/online_challenges/src/project_euler/problem_037/37.cpp [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+   int counter = 0, count = 0;
+   long long sum = 0, num = 11;
+   
+   while (counter != 11) 
+   {  
+       if (check_two_way_truncatable(num) == true) {
```
Comment (YashovardhanSiramdas) [inline code/online_challenges/src/project_euler/problem_037/37.cpp]: Same here

**Comment by square-1111** [comment]
@YashovardhanSiramdas Can you please review the changes? Thank You.

**Comment by square-1111** [comment]
@YashovardhanSiramdas  Thank You, Sir.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**